src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationHandlers.Device.cs

```diff
@@ -57,6 +59,36 @@ public ValueTask HandleAsync(ExtractDeviceAuthorizationResponseContext context)
                     context.Response["verification_url"] = null;
                 }
 
+                // Note: Huawei returns a non-standard "error" parameter as a numeric value,
+                // which is not allowed by OpenIddict (that requires a string).
+                //
+                // In device code responses, Huawei doesn't return a standard "verification_uri" parameter
+                // but returns a custom "verification_url" that serves the same purpose. It also doesn't return
+                // a standard "expires_in" parameter but returns a custom "expire_in" that serves the same purpose.
+                else if (context.Registration.ProviderType is ProviderTypes.Huawei)
+                {
+                    if ((JsonElement?) context.Response[Parameters.Error] is { ValueKind: JsonValueKind.Number })
+                    {
+                        context.Response[Parameters.Error] = Errors.InvalidRequest;
+                    }
+                    else
+                    {
+                        var verificationUri = context.Response["verification_url"];
+                        context.Response[Parameters.VerificationUri] = verificationUri;
```

Comment:
Given the URI they return cannot be used as-is - if you visit it without the `user_code` parameter attached, you get a JSON error (which is weird for a user-interactive endpoint... the user verification endpoint is not an API endpoint! 🤣) - it would likely be better to just use the `verification_uri_complete` you compute later as the `verification_uri` (so that both `verification_uri` and `verification_uri_complete` point to the same exact URI).